### PR TITLE
Support mocking of methods that return functions with variadic arguments

### PR DESCRIPTION
**Pull request by alec-w**
Description
-------------

Previously mocking a method such as
```
LogMethodf(level string) func(msg string, a ...interface{})
```
gave a struct with method signature
```
LogMethodf(level string) func(msg string, a []interface{})
```
this resolves it so that the generated struct fulfils the original contract.

- Fixes #[431](https://github.com/vektra/mockery/issues/431)

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

Version of Golang used when building/testing:
---------------------------------------------

- [ ] 1.11
- [ ] 1.12
- [ ] 1.13
- [ ] 1.14
- [ ] 1.15
- [X] 1.16
- [ ] 1.17
- [ ] 1.18

How Has This Been Tested?
---------------------------

Extended existing variadic fixture to cover this case and checked that generated mock was as expected.

Checklist
-----------

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by alec-w** [comment]
I'm still getting to grips with the generator code. This feels a bit of a hack (we strip the slice markers and replace them with ellipses if we have figured it is variadic, is there a way to stop that being done in the first place? would appear to require modifying `renderType` which is used in a lot of places). Also not fully clear on what generator tests to add.

There was also seeming a mock generated that didn't exist before - but since the interface exists in the example_project dir in the fixtures it seems this mock should exist?

**Comment by LandonTClipp** [comment]
Thanks for the update @alec-w . I am a bit swamped this week so I will take a look hopefully early next week.

**Comment by LandonTClipp** [comment]
@alec-w could you add a generator test to show that it generates the expected variadic function signature?

**Review by LandonTClipp — CHANGES_REQUESTED** [review]
@alec-w could you add a generator test to show that it generates the expected variadic function signature?

File: pkg/generator.go [hunk 1]
```diff
@@ -419,6 +419,10 @@ func (g *Generator) renderTypeTuple(ctx context.Context, tup *types.Tuple) strin
 		parts = append(parts, g.renderType(ctx, v.Type()))
 	}
 
+	if variadic {
+		parts[len(parts)-1] = fmt.Sprintf("...%s", parts[len(parts)-1][2:])
```
Comment (LandonTClipp) [inline pkg/generator.go:423]: I would prefer this to be in the for loop instead. Ex

```go
if variadic && i == tup.Len() - 1 {
  ...
} else {
 ...
}
```

**Comment by gliptak** [comment]
would the only outstanding action here "Changes Requested"?

**Comment by LandonTClipp** [comment]
I believe so yes, I’d have to take another look to see if there was anything else I missed.

**Comment by LandonTClipp** [comment]
Closing this as there hasn't been any activity for a while. Please ping me with a new PR, or on this one, if it ever gets updated. Thanks!

**Pull request closed without merging**